options/m_option.c

```diff
@@ -2381,6 +2381,58 @@ const m_option_type_t m_option_type_size_box = {
     .equal = geometry_equal,
 };
 
+static int parse_rect(struct mp_log *log, const m_option_t *opt,
+                      struct bstr name, struct bstr param, void *dst)
+{
+    bool is_help = bstr_equals0(param, "help");
+    if (is_help)
+        goto exit;
+
+    int w, h, x = 0, y = 0;
+    if (bstr_sscanf(param, "%dx%d+%d+%d", &w, &h, &x, &y) < 2)
+        goto exit;
+
+    struct mp_rect rc = {
+        .x0 = x,
+        .x1 = x + w,
+        .y0 = y,
+        .y1 = y + h,
+    };
+
+    if (dst)
+        *((struct mp_rect *) dst) = rc;
+
+    return 1;
+
+exit:
+    if (!is_help) {
+        mp_err(log, "Option %.*s: invalid rect: '%.*s'\n", BSTR_P(name),
+               BSTR_P(param));
+    }
+    mp_info(log, "Valid format: [W[%%]xH[%%][+X[%%]][+Y[%%]]]\n");
```

Comment:
I don't think this actually accepts percentages?

Reply:
I copied from m_geometry without thinking. fixed